Bump BSBLan to 0.6.2

liudger opened the pull request:
<!--
  You are amazing! Thanks for contributing to our project!
  Please, DO NOT DELETE ANY TEXT from this template! (unless instructed).
-->
Update python lib https://github.com/liudger/python-bsblan/compare/v0.5.19...v0.6.2
Fixed issue which caused to block the main thread when setting up bsblan component.

## Breaking change
<!--
  If your PR contains a breaking change for existing users, it is important
  to tell them what breaks, how to make it work again and why we did this.
  This piece of text is published with the release notes, so it helps if you
  write it towards our users, not us.
  Note: Remove this section if this PR is NOT a breaking change.
-->


## Proposed change
<!--
  Describe the big picture of your changes here to communicate to the
  maintainers why we should accept this pull request. If it fixes a bug
  or resolves a feature request, be sure to link to that issue in the
  additional information section.
-->


## Type of change
<!--
  What type of change does your PR introduce to Home Assistant?
  NOTE: Please, check only 1! box!
  If your PR requires multiple boxes to be checked, you'll most likely need to
  split it into multiple PRs. This makes things easier and faster to code review.
-->

- [x] Dependency upgrade
- [x] Bugfix (non-breaking change which fixes an issue)
- [ ] New integration (thank you!)
- [ ] New feature (which adds functionality to an existing integration)
- [ ] Deprecation (breaking change to happen in the future)
- [ ] Breaking change (fix/feature causing existing functionality to break)
- [ ] Code quality improvements to existing code or addition of tests

## Additional information
<!--
  Details are important, and help maintainers processing your PR.
  Please be sure to fill out additional details, if applicable.
-->

- This PR fixes or closes issue: fixes #
- This PR is related to issue: 
- Link to documentation pull request: 

## Checklist
<!--
  Put an `x` in the boxes that apply. You can also fill these out after
  creating the PR. If you're unsure about any of them, don't hesitate to ask.
  We're here to help! This is simply a reminder of what we are going to look
  for before merging your code.
-->

- [x] The code change is tested and works locally.
- [x] Local tests pass. **Your PR cannot be merged unless tests pass**
- [x] There is no commented out code in this PR.
- [x] I have followed the [development checklist][dev-checklist]
- [x] I have followed the [perfect PR recommendations][perfect-pr]
- [x] The code has been formatted using Ruff (`ruff format homeassistant tests`)
- [ ] Tests have been added to verify that the new code works.

If user exposed functionality or configuration variables are added/changed:

- [ ] Documentation added/updated for [www.home-assistant.io][docs-repository]

If the code communicates with devices, web services, or third-party tools:

- [x] The [manifest file][manifest-docs] has all fields filled out correctly.  
      Updated and included derived files by running: `python3 -m script.hassfest`.
- [x] New or updated dependencies have been added to `requirements_all.txt`.  
      Updated by running `python3 -m script.gen_requirements_all`.
- [x] For the updated dependencies - a link to the changelog, or at minimum a diff between library versions is added to the PR description.

<!--
  This project is very active and we have a high turnover of pull requests.

  Unfortunately, the number of incoming pull requests is higher than what our
  reviewers can review and merge so there is a long backlog of pull requests
  waiting for review. You can help here!
  
  By reviewing another pull request, you will help raise the code quality of
  that pull request and the final review will be faster. This way the general
  pace of pull request reviews will go up and your wait time will go down.
  
  When picking a pull request to review, try to choose one that hasn't yet
  been reviewed.

  Thanks for helping out!
-->

To help with the load of incoming pull requests:

- [ ] I have reviewed two other [open pull requests][prs] in this repository.

[prs]: https://github.com/home-assistant/core/pulls?q=is%3Aopen+is%3Apr+-author%3A%40me+-draft%3Atrue+-label%3Awaiting-for-upstream+sort%3Acreated-desc+review%3Anone+-status%3Afailure

<!--
  Thank you for contributing <3

  Below, some useful links you could explore:
-->
[dev-checklist]: https://developers.home-assistant.io/docs/development_checklist/
[manifest-docs]: https://developers.home-assistant.io/docs/creating_integration_manifest/
[quality-scale]: https://developers.home-assistant.io/docs/integration_quality_scale_index/
[docs-repository]: https://github.com/home-assistant/home-assistant.io
[perfect-pr]: https://developers.home-assistant.io/docs/review-process/#creating-the-perfect-pr

liudger committed:
chore: Update bsblan library to version 0.6.1

liudger committed:
add dataclass BSBLANConfig

remove session as bsblan has it's own session

liudger committed:
Update temperature unit handling in BSBLANClimate

liudger committed:
chore: Remove unused constant in bsblan/const.py

joostlek commented on homeassistant/components/bsblan/__init__.py:
```diff
@@ -35,22 +34,27 @@ class HomeAssistantBSBLANData:
         port=entry.data[CONF_PORT],
         username=entry.data.get(CONF_USERNAME),
         password=entry.data.get(CONF_PASSWORD),
-        session=session,
```
Why do we remove the session injector?

liudger replied:
I thought it was not needed as bsblan now created its own session. But I guess this is not correct?

joostlek replied:
The session we have is a bit more optimized and it saves us from creating a client session for every integration

joostlek reviewed (changes requested):
I see that you refactored the way you fetch a version, and I think it can be simpler.

I solved it in Withings like this

https://github.com/joostlek/python-withings/blob/main/src/aiowithings/withings.py#L58

liudger commented:
> 

It's not the version of the current package. It's the version of the firmware I retrieve from the board.

joostlek commented:
But I imagine that the logs you got were because you were doing `metadata.version()` in an async context?

joostlek commented:
But I imagine that the logs you got were because you were doing `metadata.version()` in an async context?

liudger commented:
> But I imagine that the logs you got were because you were doing `metadata.version()` in an async context?

I think that was the issue. It works now fine. I'll get the session back and see if it works out well.

joostlek commented:
But then I'd still recommend adapting the code, because currently you're starting a new executor job and stuff, while that can just be evaluated during the import of the library

liudger committed:
chore: Update python-bsblan library to version 0.6.2

liudger committed:
feat: Add async_get_clientsession to BSBLAN initialization

This commit adds the `async_get_clientsession` function to the initialization of the `BSBLAN` class in both `__init__.py` and `config_flow.py` files. This allows the `BSBLAN` instance to have its own session for making HTTP requests. This change improves the performance and reliability of the BSBLAN integration.

frenck commented:
Removed from 2024.8.2 milestone, as it cannot be merged cleanly onto the current stable branch (there are other PRs before this one on this integration only present on `dev`).
